test/bulk.test.ts

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+import mongoose, { Schema } from 'mongoose'
+import { config } from './config'
+import mongoosastic from '../lib/index'
+import { MongoosasticDocument, MongoosasticModel } from 'types'
+
+interface IBook extends MongoosasticDocument {
+	title: string
+}
+
+const BookSchema = new Schema<MongoosasticDocument>({
+	title: String
+})
+
+BookSchema.plugin(mongoosastic, {
+	bulk: {
+		size: 100,
+		delay: 1000
+	}
+})
+
+const Book = mongoose.model<IBook, MongoosasticModel<IBook>>('Book', BookSchema)
+
+describe('Bulk mode', function () {
+
+	beforeAll(async function () {
+		await config.deleteIndexIfExists(['books'])
+		await mongoose.connect(config.mongoUrl, config.mongoOpts)
+		await Book.deleteMany()
+
+		for (const title of config.bookTitlesArray()) {
+			await new Book({ title: title }).save()
+		}
+
+		const book = await Book.findOne({ title: 'American Gods' })
+		await book?.remove()
```

Comment:
Why is this optional?

Reply:
the findOne() return in that case an instance of (IBook | null). I could use an if condition to check if the response is different of "null", but in that case i'm sure it's an instance of 'IBook'. so i used ? operator. I will update it.

Reply:
I think that an assertion on `book != null` would be good in this case too.